src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -133,10 +120,16 @@ def append_user_agent(self, user_agent: str):
         Args:
             user_agent (str): The user agent information to append.
         """
-        agent = self.get(self._USER_AGENT, '')
-        user_agent = user_agent.strip()
-        if user_agent not in agent:
-            self[self._USER_AGENT] = f"{agent} {user_agent}".strip()
+        if "user_agent" in self:
+            if user_agent not in self.user_agent:
+                self.user_agent = f"{self.user_agent} {user_agent}"
+        else:
+            self.user_agent = user_agent
+        self.user_agent = self.user_agent
+
+    def delete_user_agent(self, user_agent: str):
```

Comment:
When will this be called?

Reply:
Our CI needs to delete UA and cannot affect other tests.
![image](https://github.com/microsoft/promptflow/assets/23182548/5498a6a4-e392-4e7a-9511-0db6b9cedf31)